From 9888451aac0af2e8a44f38b2972af7f125422652 Mon Sep 17 00:00:00 2001
From: Mooffie <mooffie@gmail.com>
Date: Sun, 1 Jan 2017 17:52:12 +0200
Subject: [PATCH 1/4] Ticket #3751: extfs: rpm: introduce the $RPM_QUERY_FMT
 variable.

This minor refactoring will enable us to make this helper testable.

Instead of doing:

    $RPM -qp --qf a b c

we are now doing:

    $RPM_QUERY_FMT a b c

In a future patch we'll then be able to point $RPM_QUERY_FMT to some mock
program.

Signed-off-by: Mooffie <mooffie@gmail.com>
---
 src/vfs/extfs/helpers/rpm | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/vfs/extfs/helpers/rpm b/src/vfs/extfs/helpers/rpm
index 4a01453..454a9c0 100755
--- a/src/vfs/extfs/helpers/rpm
+++ b/src/vfs/extfs/helpers/rpm
@@ -64,6 +64,7 @@ if rpm --nosignature --version >/dev/null 2>&1; then
 else
   RPM="rpm"
 fi
+RPM_QUERY_FMT="$RPM -qp --qf"
 RPM2CPIO="rpm2cpio"
 
 SED="sed"
@@ -92,7 +93,7 @@ mcrpmfs_getAllNeededTags()
 	tag_CONFLICTS=""
     fi
 
-    $RPM -qp --qf \
+    $RPM_QUERY_FMT \
 "|NAME=%{NAME}"\
 "|VERSION=%{VERSION}"\
 "|RELEASE=%{RELEASE}"\
@@ -113,7 +114,7 @@ ${tag_CONFLICTS}\
 
 mcrpmfs_getRawOneTag()
 {
-    $RPM -qp --qf "$1" "${rpm_filename}"
+    $RPM_QUERY_FMT "$1" "${rpm_filename}"
 }
 
 mcrpmfs_getOneTag()
-- 
2.9.3

